### PR TITLE
axi_adc_core: add `axi_adc_pn_mon` implementation

### PR DESCRIPTION
**Pull request by amiclaus**
Add implementation for pseudo-random noise sequence monitoring.

Function is required/used within multiple projects implementations.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

File: drivers/axi_core/axi_adc_core/axi_adc_core.c [hunk 1]
```diff
@@ -90,6 +90,38 @@ int32_t axi_adc_set_pnsel(struct axi_adc *adc,
...
+	mdelay(1);
+
+	for (ch = 0; ch < adc->num_channels; ch++) {
+		axi_adc_write(adc, AXI_ADC_REG_CHAN_STATUS(ch), 0xff);
+	}
+	mdelay(100);
+
+	for (ch = 0; ch < adc->num_channels; ch++) {
+		axi_adc_read(adc, AXI_ADC_REG_CHAN_STATUS(ch), &reg_data);
+		if (reg_data != 0) {
```
Comment (adrimbarean) [inline drivers/axi_core/axi_adc_core/axi_adc_core.c]: nitpick: no need for brackets

File: drivers/axi_core/axi_adc_core/axi_adc_core.c [hunk 2]
```diff
@@ -90,6 +90,38 @@ int32_t axi_adc_set_pnsel(struct axi_adc *adc,
...
+		reg_data |= AXI_ADC_ENABLE;
+		axi_adc_write(adc, AXI_ADC_REG_CHAN_CNTRL(ch), reg_data);
+		axi_adc_set_pnsel(adc, ch, sel);
+	}
+	mdelay(1);
+
+	for (ch = 0; ch < adc->num_channels; ch++) {
+		axi_adc_write(adc, AXI_ADC_REG_CHAN_STATUS(ch), 0xff);
+	}
+	mdelay(100);
```
[the remark this answers is not in the document]
Comment (amiclaus) [reply drivers/axi_core/axi_adc_core/axi_adc_core.c]: the delay is now added as parameter to the function and will be controlled in the upper layers of the applications.

File: drivers/axi_core/axi_adc_core/axi_adc_core.c (hunk 1 above)
Comment (amiclaus) [reply drivers/axi_core/axi_adc_core/axi_adc_core.c]: done

**Comment by amiclaus** [comment]
V2:

- remove unnecessary brackets
- add delay as function parameter for controlling the delay between read/write operations on the channel status registers.

**Comment by amiclaus** [comment]
V3:

- fix code style

**Pull request merged**